### PR TITLE
[postgresql] upgrade to postgresql 15

### PR DESCRIPTION
**Pull request by kayiwa**
- remove all stand-alone replication references
- set default to postgresql 15
- add new vm for postgresql staging
- remove references in tasks to replication

### CODE REVIEW

File: group_vars/postgresql/production.yml [hunk 1]
```diff
@@ -1,46 +1,43 @@
...
     priv: "ALL"
     role_attr_flags: "SUPERUSER"
-  - name: replicant
-    password: "{{ vault_replicant_password }}"
-    db: "replicant"
-    priv: "ALL"
-    role_attr_flags: "SUPERUSER"
-postgres_leader: "{{ leader_db_ip }}"
-postgres_standby: "{{ standby_db_ip }}"
+  - name: datadog
```
Comment (carolyncole) [inline group_vars/postgresql/production.yml:19]: This seem like a strange addition?

File: group_vars/postgresql/production.yml [hunk 2]
```diff
@@ -1,46 +1,43 @@
...
-postgres_host: "{{ leader_db_ip }}"
 postgres_port: 5432
 postgres_admin_user: "{{ postgresadmin }}"
 
 datadog_api_key: "{{ vault_datadog_key }}"
 datadog_config:
   log_enabled: true
 datadog_typed_checks:
-  - type: file
+  - type: postgres
```
Comment (carolyncole) [inline group_vars/postgresql/production.yml:32]: 🎉

File: roles/figgy/molecule/default/verify.yml [hunk 3]
```diff
@@ -21,9 +21,3 @@
     failed_when:
       - not figgy_mount.stat.exists
       - not figgy_mount.stat.isreg
-  - name: test for cogeo-mosaic command
```
Comment (carolyncole) [inline roles/figgy/molecule/default/verify.yml:24]: Is this no longer needed?

File: roles/postgresql/tasks/create_db.yml [hunk 4]
```diff
@@ -20,7 +20,7 @@
     state: "present"
     login_user: "{{ postgres_admin_user }}"
     owner: "{{ item.owner | default(postgres_admin_user) }}"
-  delegate_to: '{{ postgres_host }}'
+  #  delegate_to: '{{ postgres_host }}'
```
Comment (carolyncole) [inline roles/postgresql/tasks/create_db.yml:23]: Is this no longer needed for creating databases from other roles?

**Review by carolyncole — COMMENTED** [review]
I'm a bit concerned about approving this PR.  Does this mean all the applications in the PR are running jammy?  If not and the servers explode will be be able to get them back up and running on bionic?

A few other random questions.  Not trying to block you...

File: group_vars/postgresql/production.yml (hunk 1 above)
Comment (kayiwa) [reply group_vars/postgresql/production.yml:19]: you need to create a datadog user to monitor postgresql.

File: roles/postgresql/tasks/create_db.yml (hunk 4 above)
Comment (kayiwa) [reply roles/postgresql/tasks/create_db.yml:23]: that is passed using the create_db module here. https://github.com/pulibrary/princeton_ansible/blob/5168670a3909a28377bce1dc76d773d32cbe75ba/roles/postgresql/tasks/create_db.yml#L6

File: roles/figgy/molecule/default/verify.yml (hunk 3 above)
Comment (hackartisan) [reply roles/figgy/molecule/default/verify.yml:24]: ?

**Review by carolyncole — APPROVED** [review]

**Pull request merged**